### PR TITLE
Rebuild the page to make it mobile friendly.

### PR DESCRIPTION
**Pull request by mateusz-sz**
- Introduce tabbed mobile view (on small screens code editor or circuit simulation takes the whole viewport width).
- Move "Synthetize" button to top.
- Refactor and split .scss into module files.

### CODE REVIEW

**Comment by mateusz-sz** [comment]
This PR might seem big, most of the .scss changes is just splitting code into separate files.

All the other changes are bound to rebuilding the logic behind rendering the whole page, and I couldn't find a way to split the changes more.

File: src/client/index.js [hunk 1]
```diff
@@ -488,7 +489,7 @@ function runquery() {
 $('button[type=submit]').click(e => {
     e.preventDefault();
     $('#synthesize-bar .query-alert').removeClass('fade').alert('close');
-    $('form').find('input, textarea, button, select').prop('disabled', true);
```
Comment (tilk) [inline src/client/index.js:491]: What is the reason for this change?

File: src/client/index.js [hunk 2]
```diff
@@ -608,3 +613,33 @@ $('[data-bs-toggle="tooltip"]').tooltip();
 
 });
 
+
+let isAnyTabOpen = false;
```
Comment (tilk) [inline src/client/index.js:617]: You never set this back to `false`. Is this needed? Why not have one of the tabs always opened, even though it's not visible in "desktop" mode?

File: src/client/index.js [hunk 3]
```diff
@@ -608,3 +613,33 @@ $('[data-bs-toggle="tooltip"]').tooltip();
...
+const circuitTabId = 'circuit-tab';
+const codeTabBtn = $('#editor-tab-btn');
+const circuitTabBtn = $('#circuit-tab-btn');
+
+function openTab(tabId, element) {
+  $('.tab-wrapper').removeClass('open');
+  $(`#${tabId}`).addClass('open');
+
+  $('.tab-btn').removeClass('active');
+  $(`#${tabId}-btn`).addClass('active');
```
Comment (tilk) [inline src/client/index.js:628]: There is some duplication here. Why not refrain from using ids, have some classes for both tabs and buttons, and make the change in one go?

File: src/client/index.js [hunk 4]
```diff
@@ -34,20 +35,22 @@ const examples = [
 
 $(window).on('load', () => {
 
+$('#circuit-icon').append(circuitIconSvg);
```
Comment (tilk) [inline src/client/index.js:38]: Doing this in code feels ugly, IMHO. Why not get the SVG inlined in HTML or CSS? There are solutions for both of these possibilities (see, for example, https://stackoverflow.com/questions/69408473/how-do-you-selectively-inline-an-svg-in-css-with-webpack-5).

And while you're at it, why not inline the logo, too?

**Review by tilk — CHANGES_REQUESTED** [review]
Overall, this looks pretty good, but there are some minor things that should be addressed to make this change cleaner.

File: src/client/index.js (hunk 3 above)
Comment (mateusz-sz) [reply src/client/index.js:628]: Id's are needed. Otherwise I am not able to guess which button was clicked and which tab to open. How am I supposed to check which of the two buttons was clicked if they share the same class?

Comment (mateusz-sz) [reply src/client/index.js:628]: I think... I could simplify it this way, that the button and tab get the `active` class at the same time 🤔

File: src/client/index.js (hunk 2 above)
Comment (mateusz-sz) [reply src/client/index.js:617]: Actually yes - I could initially open one, arbitrary selected tab.
But the flag is needed for another purpose - on window resize we want to open the tab, that should be open.

We surely don't want to change the tab to the "default one" on every resize when screen is less than 925px wide.
Here we check it to prevent this behavior:
`window.onresize = () => {
    if (window.innerWidth <= 925 && !isAnyTabOpen) {
        openTab(codeTabClass);
    }
};`

Comment (tilk) [reply src/client/index.js:617]: Why do you want to open a tab when resizing? Why can't it be already open? This way you could throw away the entire onresize callback.

File: src/client/index.js (hunk 1 above)
Comment (mateusz-sz) [reply src/client/index.js]: Oh... I must've made a mistake. I changed it back with the last commit.

File: src/client/index.js (hunk 2 above)
Comment (mateusz-sz) [reply src/client/index.js]: Yes you were right... I noticed it after a while of thinking what an overkill this solution was. I simplified it in the last commit.

Is it good now?

File: src/client/index.js (hunk 4 above)
Comment (mateusz-sz) [reply src/client/index.js:38]: I tried to inline this SVG using method described in this stackoverflow post – unfortunately it didn't work for me, and I couldn't figure out why. I've managed to render SVG as a background image in .scss of course. But couldn't force webpack to inline it.

I don't like the solution with appending the SVG to the div, but I give up and don't have any idea how it might get inlined the other way.

Comment (mateusz-sz) [reply src/client/index.js]: I refactored it once again and successfully inline'ed the circuit button icon and also the DigitalJS logo in this commit: 03d5fb06668d1f3e4627c00f7b020dd1912b19db

File: src/client/index.js [hunk 5]
```diff
@@ -499,6 +497,10 @@ $('button[type=submit]').click(e => {
         reader.readAsText(file);
     }
     if (filenum == 0) runquery();
+
+    if (window.innerWidth <= 925) {
+        openTab(circuitTabClass);
+    }
```
Comment (tilk) [inline src/client/index.js:503]: Why do you insist on checking the window size? You can call openTab always. On desktop layout, this does nothing; on mobile, it does the right thing. The check is not needed, and the magic number in the condition introduces a risk of accidentally making a discrepancy between this and CSS later in the development.

Comment (mateusz-sz) [reply src/client/index.js]: Right. I overlooked this fact. I just pushed a quick fix to this and removed the redundant if statement.

**Review by tilk — APPROVED** [review]
